### ISSUE TITLE
Refactor Publisher to functional pattern

### ISSUE THREAD

**Issue by MovGP0** [davezych/shience#16]
The publisher should not be a [Functor](https://en.wikipedia.org/wiki/Function_object) (= class with single method; very old school), but a function (modern functional pattern). 

Instead of: 
```
Shience.Shience.SetPublisher(new MyPublisher());
```

The user should write something like: 
```
Shience.Shience.SetPublisher(result => Log.Warn(...));
Shience.Shience.SetPublisher(MyPublisher.Publish);
```

The interface `IPublisher` would become obsolete.

**Comment by davezych** [comment]
:+1: 

So something like:

    void SetPublisher(Action publish)

?

**Comment by davezych** [comment]
Implemented in #13. Closing.
